src/graphql/fetchAllRequestIdentifiers.ts

```diff
@@ -1,65 +1,103 @@
-import { IdentifierType } from '@transcend-io/privacy-types';
 
-export interface RequestIdentifier {
+const literalUnion = <T extends t.Mixed>(
+  values: T[],
+): t.UnionC<[T, T, ...T[]]> => t.union(values as [T, T, ...T[]]);
+
+const IdentifierTypeValues = literalUnion(
+  Object.values(IdentifierType).map((v) => t.literal(v)),
+);
+
+const RequestIdentifier = t.type({
   /** ID of request */
-  id: string;
+  id: t.string,
   /** Name of identifier */
-  name: string;
+  name: t.string,
   /** The underlying identifier value */
-  value: string;
-  /** Identifier metadata */
-  identifier: {
-    /** Type of identifier */
-    type: IdentifierType;
-  };
+  value: t.string,
+  /** Type of identifier */
+  type: IdentifierTypeValues,
   /** Whether request identifier has been verified at least one */
-  isVerifiedAtLeastOnce: boolean;
+  isVerifiedAtLeastOnce: t.boolean,
   /** Whether request identifier has been verified */
-  isVerified: boolean;
-}
+  isVerified: t.boolean,
+});
+
+/** Type override */
+export type RequestIdentifier = t.TypeOf<typeof RequestIdentifier>;
 
 const PAGE_SIZE = 50;
 
+export const RequestIdentifiersResponse = t.type({
+  identifiers: t.array(RequestIdentifier),
+});
+
 /**
  * Fetch all request identifiers for a particular request
  *
- * @param client - GraphQL client
- * @param options - Filter options
+ * @param options - Options
  * @returns List of request identifiers
  */
-export async function fetchAllRequestIdentifiers(
-  client: GraphQLClient,
-  {
-    requestId,
-  }: {
-    /** ID of request to filter on */
-    requestId: string;
-  },
-): Promise<RequestIdentifier[]> {
+export async function fetchAllRequestIdentifiers(client: GraphQLClient, sombra: Got, {
+  requestId,
+  decrypt,
+}: {
+  /** ID of request to filter on */
+  requestId: string;
+  /** Whether or not to decrypt identifier values */
+  decrypt: boolean;
+}): Promise<RequestIdentifier[]> {
   const requestIdentifiers: RequestIdentifier[] = [];
   let offset = 0;
-
-  // Paginate
   let shouldContinue = false;
+
   do {
-    const {
-      requestIdentifiers: { nodes },
+    let nodes: RequestIdentifier[] = [];
+
+    if (decrypt) {
+      // Get decrypted identifiers via Sombra
       // eslint-disable-next-line no-await-in-loop
-    } = await makeGraphQLRequest<{
-      /** Request Identifiers */
-      requestIdentifiers: {
-        /** List */
-        nodes: RequestIdentifier[];
-      };
-    }>(client, REQUEST_IDENTIFIERS, {
-      first: PAGE_SIZE,
-      offset,
-      requestId,
-    });
+      const response = await sombra!
+        .post<{
+          /** Decrypted identifiers */
+          identifiers: RequestIdentifier[];
+        }>('v1/request-identifiers', {
+          json: {
+            first: PAGE_SIZE,
+            offset,
+            requestId,
+          },
+        })
+        .json();
+      ({ identifiers: nodes } = decodeCodec(
+        RequestIdentifiersResponse,
+        response,
+      ));
```

Comment:
let's wrap this in a try catch and handle the case where the endpoint doesn't exist gracefully. Maybe check the error code (404 right?) and then throw a pretty message like `Transcend CLI can't reach the Sombra endpoint to decrypt the identifiers. Please ensure Sombra is the correct version to use this feature.`.

Bonus points: I think it would be even better if we could check their sombra version first but I think we'd have to make another query to sombra to do that. However, if we make the version query once at the start to find which version it's on, we could then auto-switch between these endpoints based on if they've upgraded sombra or not. That would make this super low effort to use from the customer's perspective but I'll leave it up to you if you want to implement it or not

Reply:
Oh lol looks like mike had the same idea as my bonus points: https://github.com/transcend-io/cli/pull/318#pullrequestreview-1992777981